org.lflang/src/org/lflang/generator/GeneratorCommandFactory.java

```diff
@@ -0,0 +1,144 @@
+     *
+     * @param cmd  the command to look up
+     * @param args a list of arguments
+     * @return an LFCommand object or null if the command could not be found
+     * @see #createCommand(String, List, Path, Boolean)
+     */
+    public LFCommand createCommand(String cmd, List<String> args) {
+        return createCommand(cmd, args, null, true);
+    }
+
+
+    /**
+     * Create a LFCommand instance form a given command, an argument list and a directory.
+     *
+     * @param cmd         the command to look up
+     * @param args        a list of arguments
+     * @param failOnError If true, an error is shown if the command cannot be found. Otherwise, only a warning is
+     *                    displayed.
+     * @return an LFCommand object or null if the command could not be found
+     * @see #createCommand(String, List, Path, Boolean)
+     */
+    public LFCommand createCommand(String cmd, List<String> args, Boolean failOnError) {
+        return createCommand(cmd, args, null, failOnError);
+    }
+
+
+    /**
+     * Create a LFCommand instance form a given command, an argument list and a directory.
+     * <p>
+     * The command will be executed in the CWD and if the command cannot be found an error message is shown.
+     *
+     * @param cmd  the command to look up
+     * @param args a list of arguments
+     * @param dir  the directory to execute the command in. If null, this will default to the CWD
+     * @return an LFCommand object or null if the command could not be found
+     * @see #createCommand(String, List, Path, Boolean)
+     */
+    public LFCommand createCommand(String cmd, List<String> args, Path dir) {
+        return createCommand(cmd, args, dir, true);
+    }
+
+
+    /**
+     * Create a LFCommand instance form a given command, an argument list and a directory.
+     * <p>
+     * This will check first if the command can actually be found and executed. If the command is not found, null is
+     * returned. In addition, an error message will be shown if failOnError is true. Otherwise, a warning will be
+     * displayed.
+     *
+     * @param cmd         the command to look up
+     * @param args        a list of arguments
+     * @param dir         the directory to execute the command in. If null, this will default to the CWD
+     * @param failOnError If true, an error is shown if the command cannot be found. Otherwise, only a warning is
+     *                    displayed.
+     * @return an LFCommand object or null if the command could not be found
+     * @see LFCommand#get(String, List, Path)
+     */
+    public LFCommand createCommand(String cmd, List<String> args, Path dir, Boolean failOnError) {
+        assert cmd != null && args != null;
+        if (dir == null) {
+            dir = Paths.get("");
+        }
+        LFCommand command = LFCommand.get(cmd, args, dir);
+        if (command != null) {
+            command.setEnvironmentVariable("LF_CURRENT_WORKING_DIRECTORY", dir.toString());
```

Comment:
Since environment variables are likely to be used in scripts, I would probably prefer shorter names. How about `LF_CWD`, `LF_SRC`, `LF_SRC_GEN`, and `LF_BIN`?

Reply:
I think the CMake convention at least is to spell out the entire thing, e.g., [CMAKE_SOURCE_DIR](https://cmake.org/cmake/help/latest/variable/CMAKE_SOURCE_DIR.html#variable:CMAKE_SOURCE_DIR). I like `LF_SRC` better myself personally but the fully spelled out environment variables are perhaps more familiar in a CMake context.

Reply:
Ah, yes, that makes sense.